pkg/relay/ethereum/tx.go

```diff
@@ -48,49 +48,121 @@ func (c *Chain) SendMsgs(msgs []sdk.Msg) ([]core.MsgID, error) {
 		opts.NoSend = true
 		tx, err = c.SendTx(opts, msg, skipUpdateClientCommitment)
 		if err != nil {
-			logger.Error("failed to send msg / NoSend: true", err, "msg_index", i)
+			logger.Error("failed to build tx for gas estimation", err, logAttrMsgIndex, i)
 			return nil, err
 		}
+		var rawTxData string
+		if bz, err := tx.MarshalBinary(); err != nil {
+			logger.Error("failed to encode tx", err, logAttrMsgIndex, i)
+		} else {
+			rawTxData = hex.EncodeToString(bz)
+		}
 		estimatedGas, err := c.client.EstimateGasFromTx(ctx, tx)
 		if err != nil {
-			revertReason, rawErrorData, err2 := c.getRevertReasonFromEstimateGas(err)
-			if err2 != nil {
-				logger.Error("failed to get revert reason", err2, "msg_index", i)
+			var revertReason, rawErrorData string
+			if reason, data, err := c.getRevertReasonFromEstimateGas(err); err != nil {
```

Comment:
@siburu We should include raw error data in the log even if the `ErrorRepository's parser function returns an error.

Reply:
I'm very sorry ... I fixed it.
https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/44/commits/f0ac44f10c48971e21a63b10ecfa2e02ea2f268a